roofit/roofitcore/inc/RooArgList.h

```diff
@@ -33,14 +34,31 @@ class RooArgList : public RooAbsCollection {
   ///   - RooFit objects deriving from RooAbsArg.
   ///   - A c-string to name the set.
   template<typename... Arg_t>
-  RooArgList(const RooAbsArg& arg, const Arg_t&... argsOrName)
+  RooArgList(RooAbsArg const& arg, Arg_t &&... argsOrName)
   /*NB: Making this a delegating constructor led to linker errors with MSVC*/ {
+
+    // This constructor should cause a failed static_assert if any of the input
+    // arguments is a temporary (r-value reference), which will be checked in
+    // processArg. This works statically because of the universal reference
+    // mechanism with templated functions.
+    // Unfortunately, we can't check the first arg, because it's type can't be
+    // a template parameter and hence a universal reference can't be used.
+    // This problem is solved by introducing another templated constructor below,
+    // which accepts a RooAbsArg && as the first argument which is forwarded to
+    // be the second argument for this constructor.
+
     processArg(arg);
     // Expand parameter pack in C++ 11 way:
-    int dummy[] = { 0, (processArg(argsOrName), 0) ... };
+    int dummy[] = { 0, (processArg(std::forward<Arg_t>(argsOrName)), 0) ... };
     (void)dummy;
   }
 
+  /// This constructor will provoke a `static_assert`, because passing a
+  /// RooAbsArg as r-value reference is not allowed.
+  template<typename... Arg_t>
+  RooArgList(RooAbsArg && arg, Arg_t &&... argsOrName)
+    : RooArgList{arg, std::move(arg), std::forward<Arg_t>(argsOrName)...} {}
```

Comment:
Really `std::move(arg)`, not `forward`?

Reply:
I would stay `std::move`, because here the `RooAbsArg && arg` is not a universal reference but an r-value reference.

I I think the more general `std::forward` is intended to forward universal references correctly, but here I explicitly create a new r-value reference from `arg`, which is what `move` does. Of course they both work, if you want I can change :)

Reply:
(rebasing and solving conflicts in this PR at the moment)

Reply:
Oh, so you want temporaries all the way in the arguments? That might work indeed.